README.md

```diff
@@ -14,16 +14,103 @@ When deploying an app you may need to deploy additional services, this Github Ac
-| `client_payload`      | False      | `{}`        | JSON payload with extra information about the webhook event that your action or worklow may use. |
+| `inputs`  | False       | `{}`         | Inputs to pass to the workflow, must be a JSON string |
+| `propagate_failure`      | False      | `true`        | Fail current job if downstream job fails. |
+| `trigger_workflow`       | False      | `true`        | Trigger the specified workflow. |
+| `wait_workflow`          | False      | `true`        | Wait for workflow to finish. |
 
 
 ## Example
 
+### Simple
+
+```
+- uses: convictional/trigger-workflow-and-wait@v1.3.0
+  with:
+    owner: keithconvictional
+    repo: myrepo
+    github_token: ${{ secrets.GITHUB_PERSONAL_ACCESS_TOKEN }}
+```
+
+### All Options
+
+```
+- uses: convictional/trigger-workflow-and-wait@v1.3.0
+  with:
+    owner: keithconvictional
+    repo: myrepo
+    github_token: ${{ secrets.GITHUB_PERSONAL_ACCESS_TOKEN }}
+    workflow_file_name: main.yml
+    ref: release-branch
+    wait_interval: 10
+    inputs: '{}'
+    propagate_failure: false
+    trigger_workflow: true
+    wait_workflow: true
+```
+
+
+## Testing
+
+You can test out the action locally by cloning the repository to your computer. You can run:
+
+```
+INPUT_WAITING_INTERVAL=10 \
+  INPUT_PROPAGATE_FAILURE=false \
+  INPUT_TRIGGER_WORKFLOW=true \
+  INPUT_WORKFLOW_FILE_NAME="main.yml" \
+  INPUT_WAIT_WORKFLOW=true \
+  INPUT_OWNER="keithconvictional" \
+  INPUT_REPO="trigger-workflow-and-wait-example-repo1" \
+  INPUT_GITHUB_TOKEN="<REDACTED>" \
+  INPUT_INPUTS='{}' \
+  bash entrypoint.sh
+```
+
+You will have to create a Github Personal access token. You can create a test workflow to be executed. In a repository, add a new `main.yml` to `.github/workflows/`. The workflow will be:
+
+```
+name: Main
+on:
+  workflow_dispatch
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: Pause for 25 seconds
+        run: |
+          sleep 25
+```
+
+You can see the example [here](https://github.com/keithconvictional/trigger-workflow-and-wait-example-repo1/blob/master/.github/workflows/main.yml). For testing a failure case, just add this line after the sleep:
+
+```
+...
+- name: Pause for 25 seconds
+  run: |
+    sleep 25
+    echo "For testing failure"
+    exit 1
+```
+
+
+## Potential Issues
+
+### Timing
+
+The actions dispatch is an asynchronous job and it at times can take a few seconds to start. If you do not have a delay, it may be started after you have check if it was successful. ie. Start dispatch call --> No delay --> Check if successful --> Actually starts. If the workflow has run before, it will just complete immediately as a successful run. You can solve this by simply increasing the delay to a few seconds. By default it is 10 seconds. This is also large delays between checks which helps the traffic to the Github API.
```

Comment:
nit typo: "after you have checked..." instead of "after you have check"

Reply:
nit typo: "There are also large delays.." instead of "This is also large delays"